### DISCUSSION TITLE
hydration error when refreshing the weeklys page

### DISCUSSION THREAD

**Discussion by JakobRaap** [neuefische/questions#593] [Web] [answered]
Hello, i want to display a date variable in a h3 on the weekly events page but every time i refresh the page i get a hydration errror:
`
Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.
Warning: Expected server HTML to contain a matching text node for "Monday " in h3.
`

Link to my pullrequest:
https://github.com/JakobRaap/capstone-project-economic-calendar/pull/6

```
 const mondayDate = "00-00-0000";
  const weekdayEvents = getWeekdaysEvents();
  // if (weekdayEvents.monday.length > 0) {
  //   console.log(weekdayEvents.monday[0].date);
  // }
  console.warn(events);
  return (
    <div>
      <Link href={"/"}>View todays events</Link>
      <h1>This Week&#96;s Events</h1>
      <h3>Monday {mondayDate}</h3>                    //this is where the error happens
      <WeeklyCards events={weekdayEvents.monday}></WeeklyCards>
      <h3>Tuesday</h3>
      <WeeklyCards events={weekdayEvents.tuesday}></WeeklyCards>
      <h3>Wednesday</h3>
      <WeeklyCards events={weekdayEvents.wednesday}></WeeklyCards>
      <h3>Thursday</h3>
      <WeeklyCards events={weekdayEvents.thursday}></WeeklyCards>
      <h3>Friday</h3>
      <WeeklyCards events={weekdayEvents.friday}></WeeklyCards>
    </div>
  );
```

**Comment by JessicaLoers** [comment, marked as answer]
**Hey @JakobRaap** ✌️
The error probably results from problem with the UI hydration process due to a mismatch between the server-side rendered HTML and the client-side generated HTML.

Accordingly, when you refresh the page the hydration error occurs because the events variable is undefined or empty, adding a conditional return statement (or early return)  might help. Here's an example of how to implement the check at the top of your component.

```
  if (!events || events.length === 0) {
    return;
  }
```

**Comment by JakobRaap** [comment]
thanks, that solved my problem!
